floris/core/flow_field.py

```diff
@@ -281,29 +283,45 @@ def generate_heterogeneous_wind_map(self):
                 - **y**: A list of y locations at which the speed up factors are defined.
                 - **z** (optional): A list of z locations at which the speed up factors are defined.
         """
-        speed_multipliers = self.heterogeneous_inflow_config['speed_multipliers']
+        speed_multipliers = np.array(self.heterogeneous_inflow_config['speed_multipliers'])
         x = self.heterogeneous_inflow_config['x']
         y = self.heterogeneous_inflow_config['y']
         z = self.heterogeneous_inflow_config['z']
 
+        interps_f = [] # Declare an empty list to store interpolants by findex
         if z is not None:
             # Compute the 3-dimensional interpolants for each wind direction
             # Linear interpolation is used for points within the user-defined area of values,
-            # while the freestream wind speed is used for points outside that region
-            in_region = [
-                self.interpolate_multiplier_xyz(x, y, z, multiplier, fill_value=1.0)
-                for multiplier in speed_multipliers
-            ]
+            # while the freestream wind speed is used for points outside that region.
+
+            # Because the (x,y,z) points are the same for each findex, we create the triangulation
+            # once and then overwrite the values for each findex.
+
+            # Create triangulation using zeroth findex
+            interp_3d = self.interpolate_multiplier_xyz(
+                x, y, z, speed_multipliers[0], fill_value=1.0
+            )
+            # Copy the interpolant for each findex and overwrite the values
+            for findex in range(self.n_findex):
+                interp_3d.values = speed_multipliers[findex, :].reshape(-1, 1)
+                interps_f.append(copy.deepcopy(interp_3d))
```

Comment:
Related to the comment above, if `interps_f` were changed to a np.array, it would be better to initialize it to the correct size and fill values rather than append.